sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -82,7 +83,12 @@ export async function executeAtomXmlOperation(
  */
 export function serializeToAtomXmlRequest(resourceName: string, resource: any): object {
   const content: any = {};
-  content[resourceName] = resource;
+
+  // Remove key value pairs that have value as undefined. This is because the Service Bus ATOM based management API
+  // throw a 'Bad Request' error if empty tags are included in request body.
+  // Post-processing is done as the xml2js utility does not have an option to automatically exclude empty tags.
+  content[resourceName] = removeUndefinedValues(Object.assign({}, resource));
```

Comment:
This is only copying the top level, so `removeUndefinedValues` will destructively alter sub-objects.

Possibly naive question - you say this is constructed FROM user inputs - can we just avoid creating undefined values in the first place?

Reply:
Since that would involve changes to everywhere we construct the request data, updated the docs instead as discussed offline. We may be able to alternatively update core-http in future to take in an option to post-process XML output.
Overall, since the input object is created internally, the current changes are verified to be safe.

Also, linking in relevant discussion on xml2js that may be useful to refer https://github.com/Leonidas-from-XIV/node-xml2js/issues/207